.circleci/config.yml

```diff
@@ -0,0 +1,19 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/node:6.11.1
+
+    steps:
+      - checkout
+      - restore_cache:
+          key: dependency-cache-{{ checksum "package.json" }}
+
+      - run: yarn install
+
+      - save_cache:
+          key: dependency-cache-{{ checksum "package.json" }}
+          paths:
+            - node_modules
```

Comment:
When yarn downloads packages for any project, it stores them in its cache.  Then, when a project goes to install a package that's already cached, it uses the cached version rather than re-downloading.

For a single project like our CI builds, this cache is less important, but if we're building multiple branches with different dependency versions, it become important to preserve.

Simple example:
- branch 1 uses eslint 3.17.0
- branch 2 uses eslint 4.0

When branch 1 builds the first time, yarn downloads eslint 3.17.0, stores it in its cache and in node_modules.

Then when branch 2 builds the first time, yarn downloads eslint 4.0, stores it in its cache and in node_modules.

If branch 1 builds again, node_modules doesn't have the right version any more, and neither does yarn's cache (because we didn't preserve it), so it has to re-download it.

But if we've preserved yarn's cache, then it'll find eslint 3.17.0 there and copy that into node_modules.